### PR TITLE
fix links, cleanup, and a small touch on text-matching

### PR DESCRIPTION
**Pull request by jobergum**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by jobergum** [comment]
https://vespa.ai/usecases# also needs a swipe after this clean up

File: en/text-matching.html [hunk 1]
```diff
@@ -7,44 +7,56 @@
...
-<p>Below, find details on transformations to the text for text indexing and search.
-Use inspection tools to dump data from a content node,
-and query tracing to understand matching details.</p>
+<p>Below, find details on transformations to the text for text indexing and search using the quick start 
+  sample application as an example.</p>
 
 <p>
   The <em>album</em> field has <a href="reference/schema-reference.html#indexing">index</a> mode.
-  For text fields, this enables the transformations below, and increases query recall.
+  For text fields, this enables transformations of the string field to increases query recall.
```
Comment (kkraune) [inline en/text-matching.html:54]:
```suggestion
  For text fields, this enables transformations of the string field to increase query recall.
```

File: en/text-matching.html [hunk 2]
```diff
@@ -159,15 +171,15 @@ <h2 id="index-and-attribute">Index and attribute</h2>
 ["hardwired...to self-destruct"]
 </pre>
 <p>
-The most important observation is that the strings are added <em>as-is</em> to attributes.
-Hence, values are matched in full, including whitespace.
-When searching attributes, both query terms and attribute data are lowercased before matching unless the <code>match</code> setting for the field has been set to <code>cased</code>.
+The most important observation is that the strings are added <em>as-is</em> for attributes and matching considers the full value, including whitespace (no tokenization).
+
```
Comment (kkraune) [inline en/text-matching.html:175]:
```suggestion
</p>
<p>
```

**Review by kkraune — APPROVED** [review]
Great cleanup and better text search doc now :-)

**Pull request merged**